src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/experimental_detectron_generate_proposals_single_image_ref.cl

```diff
@@ -0,0 +1,280 @@
+    const INPUT0_TYPE dy = deltas[delta_idx + 1 * BOTTOM_AREA];
+    const INPUT0_TYPE d_log_w = deltas[delta_idx + 2 * BOTTOM_AREA];
+    const INPUT0_TYPE d_log_h = deltas[delta_idx + 3 * BOTTOM_AREA];
+
+    const INPUT0_TYPE score = scores[score_idx];
+
+    // width & height of box
+    const INPUT0_TYPE ww = x1 - x0 + COORDINATES_OFFSET;
+    const INPUT0_TYPE hh = y1 - y0 + COORDINATES_OFFSET;
+    // center location of box
+    const INPUT0_TYPE ctr_x = x0 + HALF_ONE * ww;
+    const INPUT0_TYPE ctr_y = y0 + HALF_ONE * hh;
+
+    // new center location according to deltas (dx, dy)
+    const INPUT0_TYPE pred_ctr_x = dx * ww + ctr_x;
+    const INPUT0_TYPE pred_ctr_y = dy * hh + ctr_y;
+    // new width & height according to deltas d(log w), d(log h)
+    const INPUT0_TYPE pred_w = exp(min(d_log_w, TO_INPUT0_TYPE(MAX_DELTA_LOG_WH))) * ww;
+    const INPUT0_TYPE pred_h = exp(min(d_log_h, TO_INPUT0_TYPE(MAX_DELTA_LOG_WH))) * hh;
+
+    // update upper-left corner location
+    x0 = pred_ctr_x - HALF_ONE * pred_w;
+    y0 = pred_ctr_y - HALF_ONE * pred_h;
+    // update lower-right corner location
+    x1 = pred_ctr_x + HALF_ONE * pred_w - COORDINATES_OFFSET;
+    y1 = pred_ctr_y + HALF_ONE * pred_h - COORDINATES_OFFSET;
+
+    // adjust new corner locations to be within the image region
+    x0 = max(ZERO, min(x0, img_W - COORDINATES_OFFSET));
+    y0 = max(ZERO, min(y0, img_H - COORDINATES_OFFSET));
+    x1 = max(ZERO, min(x1, img_W - COORDINATES_OFFSET));
+    y1 = max(ZERO, min(y1, img_H - COORDINATES_OFFSET));
+
+    // recompute new width & height
+    const INPUT0_TYPE box_w = x1 - x0 + COORDINATES_OFFSET;
+    const INPUT0_TYPE box_h = y1 - y0 + COORDINATES_OFFSET;
+
+    proposals[proposal_idx + 0] = x0;
+    proposals[proposal_idx + 1] = y0;
+    proposals[proposal_idx + 2] = x1;
+    proposals[proposal_idx + 3] = y1;
+    proposals[proposal_idx + 4] = ((MIN_SIZE <= box_w) && (MIN_SIZE <= box_h)) ? score : 0.f;
+}
+
+#    undef COORDINATES_OFFSET
+
+#endif /* EDGPSI_STAGE_0 */
+
+#ifdef EDGPSI_STAGE_1
+
+typedef struct __attribute__((__packed__)) {
+    INPUT0_TYPE x0;
+    INPUT0_TYPE y0;
+    INPUT0_TYPE x1;
+    INPUT0_TYPE y1;
+    INPUT0_TYPE score;
+} Box;
+
+inline void FUNC(swap_box)(__global Box* a, __global Box* b) {
+    const Box temp = *a;
+    *a = *b;
+    *b = temp;
+}
+
+inline int FUNC(partition)(__global Box* arr, int l, int h) {
+    INPUT0_TYPE pivotScore = arr[h].score;
+    int i = (l - 1);
+    for (int j = l; j <= h - 1; j++) {
+        if (arr[j].score > pivotScore) {
+            i++;
+            FUNC_CALL(swap_box)(&arr[i], &arr[j]);
+        }
+    }
+    FUNC_CALL(swap_box)(&arr[i + 1], &arr[h]);
+    return (i + 1);
+}
+
+inline void FUNC(bubbleSortIterative)(__global Box* arr, int l, int h) {
+    for (int i = 0; i < h - l; i++) {
+        bool swapped = false;
+        for (int j = l; j < h - i; j++) {
+            if ((arr[j].score > arr[j + 1].score)) {
```

Comment:
The arr[j].score == arr[j+1].score case is not covered. In the refernece code (src/core/reference/include/ngraph/runtime/reference/experi....hpp), the sort is being done with std::sort, 
which will be different for the platform library. 
In high probability this undefined equal case is not a problem for CPU plugin because both CPU plugin and ngraph plugin are using same sort (possibly). However in GPU we define our own sort function as here, and the inputs order is random, such "equal score case"s result might not be aligned with the std::sort in some platforms. As a result, the functional tests in multiple platforms might fail randomly. 
So in conclusion we need to make all of the plugins to be aligned with same comparator, which might need to change all of referece impl, cpu plugin kernel impl, and gpu kernel here... 
Please refer to DetectionOutput's experience. https://github.com/openvinotoolkit/openvino/pull/6530

Reply:
how do you think, should it be done in scope or this PR?

Reply:
I think it should be applied before this PR is merged.
Also it is better to be applied as a separate PR from this PR for better review process, since it will need to touch other components (ngraph, cpu plugin). 
Otherwise it will make our CI to fail randomly after this PR is merged.

Reply:
It is a common issue for all GPU operations which implements own sort. There just a few for now, NMS and DetectionOutput, and the issues are not existing for those ops because they implemented handling of same score cases. Now you are adding this new ops without handling of same score cases, we need to add that hadnling before merging this PR. Otherwise we will experience random failure of our CI in the future.

Reply:
understood. So for now  I think the best case would be to override generation as it's done in DetectionOutput. Would it be acceptable for this PR?

Reply:
actually now inputs are already received from params, but I can change it to generation

Reply:
@yeonbok  as inputs are not random and we don't expect sporadic fails because of this can we handle this conversation as resolved?

Reply:
Oh, this test is not using random input as others. Then it is okay to go with. Thanks for checking.